plugins/transport/socket/main.go

```diff
@@ -222,7 +222,7 @@ func (s *Socket) Run(ctx context.Context, w transport.WriteFn, done chan bool) {
 	switch s.conf.Type {
 	case udp:
 		pc = s.initUDPSocket()
-		if pc == nil {
+		if pc == (*net.UDPConn)(nil) {
```

Comment:
```
Error: SA4023: this comparison is never true (staticcheck)
Error: SA4023(related information): the lhs of the comparison gets its value from here and has a concrete type (staticcheck)
```

I found this one interesting because the message was scary and suggested the error handling would be broken without the cast ("this comparison is never true"). AFAICT, that's not the case: https://go.dev/play/p/3URVPk76gDP

Reply:
pc is a type of net.Conn, which is an interface, which technically (IMO) should not ever be nil. You can do such comparison with pointers as those can point to nowhere. Interfaces should be either "empty" (as you fixed that) or holding some value (even though they are using pointer under the hood IIUC) .